python-frontend/src/main/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProvider.java

```diff
@@ -128,4 +139,35 @@ static ModuleSymbol deserializedModule(String moduleName, InputStream resource)
     }
   }
 
+
+  /**
+   * Returns stub symbols to be used by SonarSecurity.
+   * Ambiguous symbols that only contain class symbols are disambiguated with latest Python version.
+   */
+  public Collection<Symbol> stubFilesSymbols(ProjectLevelSymbolTable projectLevelSymbolTable) {
+    if (cachedSymbols != null) {
+      return cachedSymbols;
+    }
+    Collection<Descriptor> allDescriptors = new ArrayList<>(stubFilesDescriptors());
+    allDescriptors.addAll(builtinDescriptors().values());
+    record Pair(Symbol symbol, Descriptor descriptor){}
+
+    Map<String, Pair> symbolsByFqn = new HashMap<>();
+    cachedSymbols = new HashSet<>();
+
+    for (Descriptor descriptor : allDescriptors) {
+      if (descriptor.fullyQualifiedName() != null) {
+        Pair pair = symbolsByFqn.computeIfAbsent(descriptor.fullyQualifiedName(), k ->
+          new Pair(DescriptorUtils.symbolFromDescriptor(descriptor, projectLevelSymbolTable, null, new HashMap<>(), new HashMap<>()), descriptor));
+        cachedSymbols.add(pair.symbol());
+      }
+    }
+    return cachedSymbols;
+  }
```

Comment:
Why do you have to introduce a record here? 
```suggestion
  public Collection<Symbol> stubFilesSymbols(ProjectLevelSymbolTable projectLevelSymbolTable) {
    if (cachedSymbols != null) {
      return cachedSymbols;
    }
    Collection<Descriptor> allDescriptors = new ArrayList<>(stubFilesDescriptors());
    allDescriptors.addAll(builtinDescriptors().values());

    Map<String, Symbol> symbolsByFqn = new HashMap<>();
    cachedSymbols = new HashSet<>();

    for (Descriptor descriptor : allDescriptors) {
      if (descriptor.fullyQualifiedName() != null) {
        Symbol symbol = symbolsByFqn.computeIfAbsent(descriptor.fullyQualifiedName(), k -> DescriptorUtils.symbolFromDescriptor(descriptor, projectLevelSymbolTable, null, new HashMap<>(), new HashMap<>()));
        cachedSymbols.add(symbol);
      }
    }
    return cachedSymbols;
  }
```

Reply:
I don't really remember why I needed that in a previous iteration...but certainly I don't anymore, good catch!